test/unit/services/test_catalogue_category.py

```diff
@@ -307,6 +356,150 @@ def test_create_with_duplicate_property_names(catalogue_category_repository_mock
+            is_leaf=False,
+            parent_id=None,
+            catalogue_item_properties=[],
+            created_time=MODEL_MIXINS_FIXED_DATETIME_NOW,
+            modified_time=MODEL_MIXINS_FIXED_DATETIME_NOW,
+        ),
+    )
+    # pylint: enable=duplicate-code
+
+    # Mock `get` to return the unit
+    test_helpers.mock_get(unit_repository_mock, None)
+
+    # Mock `create` to return the created catalogue category
+    test_helpers.mock_create(catalogue_category_repository_mock, catalogue_category)
+
+    with pytest.raises(MissingRecordError) as exc:
+        catalogue_category_service.create(
+            CatalogueCategoryPostRequestSchema(
+                name=catalogue_category.name,
+                is_leaf=catalogue_category.is_leaf,
+                parent_id=catalogue_category.parent_id,
+                catalogue_item_properties=[prop.model_dump() for prop in catalogue_category.catalogue_item_properties],
+            )
+        )
+    catalogue_category_repository_mock.create.assert_not_called()
+    assert str(exc.value) == (f"No unit found with ID: {unit.id}")
+
+
+def test_create_catalogue_item_properties_with_invalid_unit_id(
+    test_helpers,
+    catalogue_category_repository_mock,
+    unit_repository_mock,
+    model_mixins_datetime_now_mock,  # pylint: disable=unused-argument
+    catalogue_category_service,
+):
+    """
+    Test creating a catalogue category with a non existent unit ID.
+    """
+
+    unit = UnitOut(id="invalid", **UNIT_A)
+
+    # pylint: disable=duplicate-code
+    catalogue_category = CatalogueCategoryOut(
+        id=str(ObjectId()),
+        name="Category B",
+        code="category-b",
+        is_leaf=True,
+        parent_id=str(ObjectId()),
+        catalogue_item_properties=[
+            CatalogueItemPropertyOut(
+                id=str(ObjectId()),
+                name="Property A",
+                type="number",
+                unit_id=unit.id,
+                unit=unit.value,
+                mandatory=False,
+            ),
+            CatalogueItemPropertyOut(id=str(ObjectId()), name="Property B", type="boolean", mandatory=True),
+        ],
+        created_time=MODEL_MIXINS_FIXED_DATETIME_NOW,
+        modified_time=MODEL_MIXINS_FIXED_DATETIME_NOW,
+    )
+    # pylint: enable=duplicate-code
+
+    # Mock `get` to return the parent catalogue category
+    # pylint: disable=duplicate-code
+    test_helpers.mock_get(
+        catalogue_category_repository_mock,
+        CatalogueCategoryOut(
+            id=catalogue_category.parent_id,
+            name="Category A",
+            code="category-a",
+            is_leaf=False,
+            parent_id=None,
+            catalogue_item_properties=[],
+            created_time=MODEL_MIXINS_FIXED_DATETIME_NOW,
+            modified_time=MODEL_MIXINS_FIXED_DATETIME_NOW,
+        ),
+    )
+    # pylint: enable=duplicate-code
+
+    # Mock `get` to return the unit
+    test_helpers.mock_get(unit_repository_mock, None)
```

Comment:
This is just doing the same as the test above rather than checking for an InvalidObjectIdError. I realise now the test I saw is on its own and is for database integrity, so maybe this test doesn't make sense after all. I think it should just be removed again its tested elsewhere anyway, sorry 😅